tt_metal/hw/inc/ethernet/dataflow_api.h

```diff
@@ -439,18 +461,24 @@ void eth_receiver_done() {
  * |-----------------------------|---------------------------------------------------------|----------|-------------|----------|
  * | channel                     | Which transaction channel to ack                        | uint32_t | 0..7        | True     |
  */
+
 FORCE_INLINE
-void eth_receiver_channel_done(uint32_t channel) {
-    // assert(channel < 4);
-    erisc_info->channels[channel].bytes_sent = 0;
-    erisc_info->channels[channel].receiver_ack = 0;
+void send_eth_receiver_channel_done(volatile eth_channel_sync_t *channel_sync) {
+    channel_sync->bytes_sent = 0;
+    channel_sync->receiver_ack = 0;
     internal_::eth_send_packet(
         0,
-        ((uint32_t)(&(erisc_info->channels[channel].bytes_sent))) >> 4,
-        ((uint32_t)(&(erisc_info->channels[channel].bytes_sent))) >> 4,
+        ((uint32_t)(channel_sync)) >> 4,
```

Comment:
Although bytes_sent is at the start of the  struct, you should still take the address of the field so that the code will work with any struct changes.

Reply:
Was just trying to avoid that extra de-reference for perf but I guess a good optimizer will realize that it can be optimized away

Reply:
There is no derefence required since we're simply taking the address. (See here as an example: https://godbolt.org/z/91Y8E7n18). Alternatively, you could do `static_assert(offsetof(eth_channel_sync_t, bytes_sent) == 0), "assumption failed yada yada yada");`